packages/adblocker/src/filters/cosmetic.ts

```diff
@@ -260,7 +268,25 @@ export default class CosmeticFilter implements IFilter {
     // number of labels considered. This allows a compact representation of
     // hostnames and fast matching without any string copy.
     if (sharpIndex > 0) {
-      domains = Domains.parse(line.slice(0, sharpIndex), { debug });
+      const domainEntries = [];
+      const parentDomainEntries = [];
+      for (const entry of line.slice(0, sharpIndex).split(',')) {
+        if (entry.endsWith('>>')) {
+          parentDomainEntries.push(entry.slice(0, -2));
+        } else {
+          domainEntries.push(entry);
+        }
+      }
```

Comment:
The `>>` token is the suffix of domain entities. For an instance, filter can have multiple parent domain values by appending multiple `<hostname>>>` values: `foo.com>>,bar.com>>`.